### PR TITLE
- Unnecessary Lock type parameter removed from shared_guarded

### PR DESCRIPTION
**Pull request by ruszkait**
- Unnecessary Lock type parameter removed from shared_guarded. For the shared use-case at the moment I do not see any other valid option than the shared_lock.

- Unnecessary lock check removed in the handler deleter: null unique_ptr does not call the deleter (https://en.cppreference.com/w/cpp/memory/unique_ptr/%7Eunique_ptr), so the conditional statement there is always true.

- Simplified the invalid handle related code: with default constructible unique_ptr deleter, the invalid handle is easier (and maybe cheaper) to generate

### CODE REVIEW

**Comment by agserm** [comment]
Thank you for your contribution, will take a look at this later this week.

File: src/shared_guarded.hpp [hunk 1]
```diff
@@ -76,12 +76,14 @@ class shared_guarded
...
 
-template <typename T, typename M, typename L>
-class shared_guarded<T, M, L>::deleter
+template <typename T, typename M>
+class shared_guarded<T, M>::deleter
 {
   public:
     using pointer = T *;
 
+    deleter();
```
Comment (thed636) [inline src/shared_guarded.hpp:85]: =default ?

File: src/shared_guarded.hpp [hunk 2]
```diff
@@ -90,141 +92,150 @@ class shared_guarded<T, M, L>::deleter
...
+}
+
+template <typename T, typename M>
+class shared_guarded<T, M>::shared_deleter
 {
   public:
     using pointer = const T *;
 
-    shared_deleter(L lock);
+    shared_deleter();
```
Comment (thed636) [inline src/shared_guarded.hpp:117]: =default ?

File: src/guarded.hpp [hunk 3]
```diff
@@ -116,6 +116,8 @@ class guarded<T, M>::deleter
   public:
     using pointer = T *;
 
+    deleter();
```
[the remark this answers is not in the document]
Comment (ruszkait) [reply src/guarded.hpp:119]: ok

Comment (agserm) [inline src/guarded.hpp:119]: This has the side effect of allowing the user to default construct a guarded<T>::handle type. If this is a good idea, it is something which may need to be done in other classes as well.

Comment (ruszkait) [reply src/guarded.hpp:119]: I have made another push, now the default deleter is removed

**Comment by agserm** [comment]
Closing due to lack of response to comments, please feel free to reopen when comments are addressed or if further discussion is needed.

**Pull request closed without merging**